### PR TITLE
chore: Uses TestStepCheckEmptyPlan in mig tests

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Uses TestStepCheckEmptyPlan in mig tests

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals I have added appropriate changelog entries.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/backupcompliancepolicy/resource_backup_compliance_policy_test.go [hunk 1]
```diff
@@ -150,15 +149,6 @@ func TestAccBackupCompliancePolicy_UpdateSetsAllAttributes(t *testing.T) {
...
 			},
-			{
-				Config: configBasicWithOptionalAttributesWithNonDefaultValues(projectName, orgID, projectOwnerID, "8"),
-				ConfigPlanChecks: resource.ConfigPlanChecks{
-					PreApply: []plancheck.PlanCheck{
-						acc.DebugPlan(),
-						plancheck.ExpectEmptyPlan(),
-					},
-				},
-			},
```
Comment (lantoli) [inline internal/service/backupcompliancepolicy/resource_backup_compliance_policy_test.go:161]: not a mig test (not in in migration test file)

File: internal/service/cluster/resource_cluster_test.go [hunk 2]
```diff
@@ -580,15 +579,6 @@ func TestAccCluster_ProviderRegionName(t *testing.T) {
...
 			},
-			{
-				Config: configMultiRegion(orgID, projectName, clusterName, "false", updatedRegionsConfig),
-				ConfigPlanChecks: resource.ConfigPlanChecks{
-					PreApply: []plancheck.PlanCheck{
-						acc.DebugPlan(),
-						plancheck.ExpectEmptyPlan(),
-					},
-				},
-			},
```
Comment (lantoli) [inline internal/service/cluster/resource_cluster_test.go:591]: not a mig test (in in migration test file)

File: internal/service/federateddatabaseinstance/resource_federated_database_instance_test.go [hunk 3]
```diff
@@ -63,15 +62,6 @@ func TestAccFederatedDatabaseInstance_basic(t *testing.T) {
...
-				Config: configFirstStepsUpdate(name, projectName, orgID),
-				ConfigPlanChecks: resource.ConfigPlanChecks{
-					PreApply: []plancheck.PlanCheck{
-						acc.DebugPlan(),
-						plancheck.ExpectEmptyPlan(),
-					},
-				},
-			},
 		},
 	})
```
Comment (lantoli) [inline internal/service/federateddatabaseinstance/resource_federated_database_instance_test.go:66]: not a mig test (in in migration test file)

File: internal/service/project/resource_project_test.go [hunk 4]
```diff
@@ -1045,15 +1044,6 @@ func TestAccProject_withTags(t *testing.T) {
...
 			},
-			{
-				Config: configWithTags(orgID, projectName, tagsOnlyIgnored),
-				ConfigPlanChecks: resource.ConfigPlanChecks{
-					PreApply: []plancheck.PlanCheck{
-						acc.DebugPlan(),
-						plancheck.ExpectEmptyPlan(),
-					},
-				},
-			},
```
Comment (lantoli) [inline internal/service/project/resource_project_test.go:1056]: not a mig test (in in migration test file)

**Review by EspenAlbert — APPROVED** [review]
Awesome!

**Pull request merged**